apps/dashboard/src/app/team/[team_slug]/(team)/~/engine/(instance)/[engineId]/webhooks/components/webhooks-table.tsx

```diff
@@ -113,36 +119,86 @@ export const WebhooksTable: React.FC<WebhooksTableProps> = ({
   isPending,
   isFetched,
 }) => {
-  const [webhookToRevoke, setWebhookToRevoke] = useState<EngineWebhook>();
-  const { isOpen, onOpen, onClose } = useDisclosure();
-  const { mutate: revokeWebhook } = useEngineRevokeWebhook(instanceUrl);
+  const [selectedWebhook, setSelectedWebhook] = useState<EngineWebhook>();
+  const deleteDisclosure = useDisclosure();
+  const testDisclosure = useDisclosure();
+
+  const activeWebhooks = webhooks.filter((webhook) => webhook.active);
+
+  return (
+    <>
+      <TWTable
+        title="webhooks"
+        data={activeWebhooks}
+        columns={columns}
+        isPending={isPending}
+        isFetched={isFetched}
+        onMenuClick={[
+          {
+            icon: <MailQuestion className="size-4" />,
+            text: "Test webhook",
+            onClick: (row) => {
+              setSelectedWebhook(row);
+              testDisclosure.onOpen();
+            },
+          },
+          {
+            icon: <TrashIcon className="size-4" />,
+            text: "Delete",
+            onClick: (row) => {
+              setSelectedWebhook(row);
+              deleteDisclosure.onOpen();
+            },
+            isDestructive: true,
+          },
+        ]}
+      />
+
+      {selectedWebhook && deleteDisclosure.isOpen && (
+        <DeleteWebhookModal
+          webhook={selectedWebhook}
+          disclosure={deleteDisclosure}
+          instanceUrl={instanceUrl}
+        />
+      )}
+      {selectedWebhook && testDisclosure.isOpen && (
+        <TestWebhookModal
+          webhook={selectedWebhook}
+          disclosure={testDisclosure}
+          instanceUrl={instanceUrl}
+        />
+      )}
+    </>
+  );
+};
+
+interface DeleteWebhookModalProps {
+  webhook: EngineWebhook;
+  disclosure: UseDisclosureReturn;
+  instanceUrl: string;
+}
+function DeleteWebhookModal({
+  webhook,
+  disclosure,
+  instanceUrl,
+}: DeleteWebhookModalProps) {
+  const { mutate: deleteWebhook } = useEngineDeleteWebhook(instanceUrl);
   const trackEvent = useTrack();
   const { onSuccess, onError } = useTxNotifications(
```

Comment:
Use `toast` instead. Only use useTxNotifications when sending transactions. I know this is not added as part of this PR - but lets change it anyway